tooling/nargo_toml/src/errors.rs

```diff
@@ -69,6 +69,9 @@ pub enum ManifestError {
 
     #[error(transparent)]
     SemverError(SemverError),
+
+    #[error("Cyclic package dependency found when processing {toml}")]
+    CyclicDependency { toml: PathBuf },
```

Comment:
Can we attempt to show the user the cyclic dependency here? I think it'd help a lot while debugging.

Reply:
The error displays the full cycle now.